### PR TITLE
Added: Support for Search API

### PR DESCRIPTION
**Pull request by pitiphong-p**

### CODE REVIEW

File: OmiseSwift/Charge.swift [hunk 1]
```diff
@@ -131,6 +136,40 @@ public class ChargeParams: Params {
...
+    }
+    
+    public var customerPresent: Bool? {
+        get { return get("customer_present", BoolConverter.self) }
+        set { set("customer_present", BoolConverter.self, toValue: newValue) }
+    }
+    
+    public var failureCode: String? {
+        get { return get("failure_code", StringConverter.self) }
+        set { set("failure_code", StringConverter.self, toValue: newValue) }
```
Comment (chakrit) [inline OmiseSwift/Charge.swift:166]: do we have `failure_message` too?

File: OmiseSwift/Charge.swift [hunk 2]
```diff
@@ -95,6 +95,11 @@ public class Charge: ResourceObject {
 }
 
 public class ChargeParams: Params {
+    public var created: NSDateComponents? {
+        get { return get("created", DateComponentsConverter.self) }
+        set { set("created", DateComponentsConverter.self, toValue: newValue) }
+    }
```
Comment (chakrit) [inline OmiseSwift/Charge.swift]: Hmm, why is this needed?

File: OmiseSwift/DateComponentsConverter.swift [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            scanner.scanInteger(&secondInt) &&
+            scanner.scanString("-", intoString: nil) &&
+            scanner.scanInteger(&lastInt) &&
+            scanner.scanString("-", intoString: nil)
+            else {
+                return nil
+        }
+        
+        month = secondInt
+        let isYearFirst = firstInt > 12
```
Comment (chakrit) [inline OmiseSwift/DateComponentsConverter.swift:34]: I think we should just fix the format here.

Comment (chakrit) [inline OmiseSwift/DateComponentsConverter.swift:34]: Always go year first.

**Comment by chakrit** [comment]
Only some minor issues.

File: OmiseSwift/Charge.swift (hunk 2 above)
Comment (pitiphong-p) [reply OmiseSwift/Charge.swift]: Wrong place, will fix now

File: OmiseSwift/DateComponentsConverter.swift (hunk 3 above)
Comment (pitiphong-p) [reply OmiseSwift/DateComponentsConverter.swift:34]: OK.

Comment (pitiphong-p) [reply OmiseSwift/DateComponentsConverter.swift:34]: Search API Interface supports both format. Should we also support both format too?

File: OmiseSwift/Charge.swift (hunk 1 above)
Comment (pitiphong-p) [reply OmiseSwift/Charge.swift:166]: `failure_message` isn't in the supported filter list.

File: OmiseSwift/DateComponentsConverter.swift (hunk 3 above)
Comment (chakrit) [reply OmiseSwift/DateComponentsConverter.swift:34]: No need, it should just be able to parse what we ourselves send.

**Comment by chakrit** [comment]
👍
